pkg/controller/hosted/controller.go

```diff
@@ -470,13 +473,23 @@ func createHostingManifestWork(managedClusterName string,
 		if err != nil {
 			panic(err)
 		}
+
+		unstructuredObj := &unstructured.Unstructured{}
+		if err := json.Unmarshal(jsonData, unstructuredObj); err != nil {
+			klog.Errorf("failed to unmarshal object %s: %v", string(jsonData), err)
+			continue
+		}
+		// the namespace has been created, should not be managed by manifestWork.
+		if unstructuredObj.GetKind() == "Namespace" {
+			continue
+		}
 		manifests = append(manifests, workv1.Manifest{
 			RawExtension: runtime.RawExtension{Raw: jsonData},
 		})
 	}
 
-	// For hosted mode, the klusterletManifestWork only contains a klusterlet CR
-	// and a bootstrap secret, delete it in foreground.
+	// For hosted mode, the klusterletManifestWork only contains a klusterlet CR, a bootstrap secret and an agent namespace,
+	// delete klusterlet and bootstrap in foreground, orphan the agent namespace.
```

Comment:
It seems that the namespace is excluded above?

Reply:
deleted the wrong comment.